### PR TITLE
Lots of different, smaller fixes

### PR DESCRIPTION
**Pull request by Trigary**
code cleanup, improved exception handling, fixed typos, removal of redundant stuff, less code duplication, maven version bumps

In case any of the changes are questionable or seem off: I would be happy to answer any questions, explain myself.

### CODE REVIEW

File: src/main/java/com/github/stefvanschie/inventoryframework/util/SkullUtil.java [hunk 1]
```diff
@@ -38,22 +38,29 @@ private SkullUtil() {}
...
+        item.setItemMeta(itemMeta);
+        return item;
+    }
 
+    /**
+     * Sets the skull of an existing {@link ItemMeta} from the specified id.
+     * The id is the value from the textures.minecraft.net website after the last '/' character.
+     *
+     * @param id the skull id
+     */
```
Comment (stefvanschie) [inline src/main/java/com/github/stefvanschie/inventoryframework/util/SkullUtil.java]: Please provide an `@param` annotation for the `meta` parameter.

File: src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java [hunk 2]
```diff
@@ -23,34 +23,36 @@
...
                         method.invoke(instance, event);
-                    } catch (IllegalAccessException | InvocationTargetException e) {
-                        e.printStackTrace();
+                    } else {
+                        method.invoke(instance);
                     }
-                };
-            }
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                    throw new RuntimeException("Error while invoking reflection-obtained event listener", e);
```
Comment (stefvanschie) [inline src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java]: Please change this exception type to `XMLReflectionException`.

File: src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java [hunk 3]
```diff
@@ -70,7 +72,7 @@ public static void loadFieldAttribute(@NotNull Object instance, @NotNull Element
             field.setAccessible(true);
             field.set(instance, value);
         } catch (NoSuchFieldException | IllegalAccessException e) {
-            e.printStackTrace();
+            throw new RuntimeException(e);
```
Comment (stefvanschie) [inline src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java]: Please change this exception type to `XMLReflectionException`.

Comment (Trigary) [reply src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java]: This code is executed while loading XML files, so shouldn't `XMLLoadException` be used instead?

File: src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java (hunk 2 above)
Comment (Trigary) [reply src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java]: I'm dumm

File: src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java (hunk 3 above)
Comment (stefvanschie) [reply src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java]: Yeah, sorry, `XMLLoadException` not `XMLReflectionException`.

**Comment by Trigary** [comment]
I expected you to squash my commits - I wouldn't have made such a retarded commit message if I knew you wouldn't lol

**Pull request merged**